src/i2c.rs

```diff
@@ -133,16 +130,16 @@ impl<I2C, SCL, SDA> I2c<I2C, (SCL, SDA)> {
         // t_SYNC1 + t_SYNC2 > 4 * t_I2CCLK
         // t_SCL ~= t_SYNC1 + t_SYNC2 + t_SCLL + t_SCLH
         let i2cclk = I2C::clock(&clocks).0;
-        let ratio = i2cclk / freq - 4;
-        let (presc, scll, sclh, sdadel, scldel) = if freq >= 100_000 {
+        let ratio = i2cclk / freq.0 - 4;
```

Comment:
Seems like `embedded-time` has an [`integer`](https://docs.rs/embedded-time/0.10.1/embedded_time/rate/trait.Rate.html#get-the-integer-part) method, which could be used instead. 

```suggestion
        let ratio = i2cclk / freq.integer() - 4;
```

Make the code a little better to understand but also more verbose. I'm fine, if you want to keep it as is for now.